### PR TITLE
chore(e2e): browserstack

### PR DESCRIPTION
**Pull request by frassinier**
**What is the problem this PR is trying to solve?**
Tests end to end (e2e) are not executed from CI

**What is the chosen solution to this problem?**
Add Browserstack support
Execute them all for each pull-request

**Please check if the PR fulfills these requirements**
- [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,37 +1,65 @@
...
+jobs:
+  include:
+    - stage: build
+      env:
+        - ACTION=test
+      script:
+        - yarn $ACTION
+    - # stage name not required, will continue to use `test`
+      env:
+        - ACTION=test:demo
```
Comment (ncomont) [inline .travis.yml]: Why did you used an env var here ?

File: .travis/after_success_e2e.sh [hunk 2]
```diff
@@ -0,0 +1,15 @@
+#!/usr/bin/env bash
+
+echo "E2E TESTS"
+cd "$TRAVIS_BUILD_DIR"
+cd packages/components/e2e/component-objects
```
Comment (ncomont) [inline .travis/after_success_e2e.sh]: `cd $TRAVIS_BUILD_DIR/packages/components/e2e/component-objects`

File: packages/components/e2e/component-objects/src/test/java/org/talend/component/StorybookTest.java [hunk 3]
```diff
@@ -1,44 +1,41 @@
...
 
     @BeforeClass
-    public static void before() {
-        driver = new ChromeDriver();
-        driver.manage().window().maximize();
+    public static void before() throws Exception {
+        webDriverTestConfiguration = new WebDriverTestFactory().getWebDriverTestConfiguration();
+        webDriverTestConfiguration.setUp();
+        driver = webDriverTestConfiguration.driver;
+        System.out.println("WebDriver is running: " + driver);
```
Comment (ncomont) [inline packages/components/e2e/component-objects/src/test/java/org/talend/component/StorybookTest.java]: Why didn't you used the `Logger` used in other classes ?

File: packages/components/e2e/component-objects/src/test/java/org/talend/config/BrowserStackWebDriverTest.java [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+    public void setUp() throws Exception {
+        final DesiredCapabilities capabilities = new DesiredCapabilities();
+        capabilities.setCapability("browser", "Chrome");
+        capabilities.setCapability("browser_version", "64.0");
+        capabilities.setCapability("os", "Windows");
+        capabilities.setCapability("os_version", "10");
+        capabilities.setCapability("resolution", "1024x768");
+        capabilities.setCapability("browserstack.debug", "true");
+        //capabilities.setCapability("browserstack.local", "true");
```
Comment (ncomont) [inline packages/components/e2e/component-objects/src/test/java/org/talend/config/BrowserStackWebDriverTest.java]: ?

File: packages/components/e2e/component-objects/src/test/java/org/talend/config/LocalWebDriverTest.java [hunk 5]
```diff
@@ -0,0 +1,16 @@
+package org.talend.config;
+
+import org.openqa.selenium.chrome.ChromeDriver;
+
+public class LocalWebDriverTest extends WebDriverTest {
+
+    public void setUp() {
+        System.out.println("Using WebDriver from " + System.getProperty("webdriver.chrome.driver"));
```
Comment (ncomont) [inline packages/components/e2e/component-objects/src/test/java/org/talend/config/LocalWebDriverTest.java]: Why didn't you used the `Logger` used in other classes ?

File: packages/components/e2e/component-objects/src/test/java/org/talend/config/WebDriverTestFactory.java [hunk 6]
```diff
@@ -0,0 +1,15 @@
+package org.talend.config;
+
+public class WebDriverTestFactory {
+
+    public WebDriverTest getWebDriverTestConfiguration() {
+        boolean isRemote = Boolean.parseBoolean(System.getProperty("remote", "false"));
+        if (isRemote) {
+            System.out.println("Remote WebDriver test configuration enabled");
```
Comment (ncomont) [inline packages/components/e2e/component-objects/src/test/java/org/talend/config/WebDriverTestFactory.java]: Why didn't you used the `Logger` used in other classes ?

File: packages/components/e2e/component-objects/src/test/java/org/talend/config/BrowserStackWebDriverTest.java (hunk 4 above)
Comment (frassinier) [reply packages/components/e2e/component-objects/src/test/java/org/talend/config/BrowserStackWebDriverTest.java]: Oh god, I forgot to deal with it

File: .travis.yml (hunk 1 above)
Comment (frassinier) [reply .travis.yml]: Waiting for some help from @jmfrancois

File: packages/components/e2e/component-objects/src/test/java/org/talend/config/LocalWebDriverTest.java (hunk 5 above)
Comment (frassinier) [reply packages/components/e2e/component-objects/src/test/java/org/talend/config/LocalWebDriverTest.java]: 👌

File: packages/components/e2e/component-objects/src/test/java/org/talend/config/WebDriverTestFactory.java (hunk 6 above)
Comment (frassinier) [reply packages/components/e2e/component-objects/src/test/java/org/talend/config/WebDriverTestFactory.java]: 👌

File: packages/components/e2e/component-objects/src/test/java/org/talend/component/StorybookTest.java (hunk 3 above)
Comment (frassinier) [reply packages/components/e2e/component-objects/src/test/java/org/talend/component/StorybookTest.java]: 👌

**Pull request merged**